packages/luigis-box/src/Component/LuigisBox/LuigisBoxClient.php

```diff
@@ -222,21 +245,42 @@ protected function addAutocompleteSpecificParametersToUrl(
     }
 
     /**
-     * @param string $url
      * @param \Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxBatchLoadData $luigisBoxBatchLoadData
-     * @param array<string, int> $limitsByType
-     * @return string
+     * @return array
      */
-    protected function addRecommendationsParameters(
-        string $url,
+    protected function getBody(
         LuigisBoxBatchLoadData $luigisBoxBatchLoadData,
-        array $limitsByType,
-    ): string {
+    ): array {
+        $body = [];
+
+        if ($luigisBoxBatchLoadData instanceof LuigisBoxRecommendationBatchLoadData) {
+            $body = $this->addRecommendationSpecificParametersToBody($body, $luigisBoxBatchLoadData);
+        }
+
+        return [$body];
+    }
+
+    /**
+     * @param array $body
+     * @param \Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxRecommendationBatchLoadData $luigisBoxBatchLoadData
+     * @return array
+     */
+    protected function addRecommendationSpecificParametersToBody(
+        array $body,
+        LuigisBoxBatchLoadData $luigisBoxBatchLoadData,
+    ): array {
         if ($luigisBoxBatchLoadData->getEndpoint() === LuigisBoxEndpointEnum::RECOMMENDATIONS) {
-            $url .= '&recommendation_type=' . $this->mapLimitsByTypeToLuigisBoxLimit($limitsByType);
+            $body['recommendation_type'] = $luigisBoxBatchLoadData->getType();
+            $body['user_id'] = $luigisBoxBatchLoadData->getUserIdentifier();
+            $body['size'] = $luigisBoxBatchLoadData->getLimit();
+            $body['hit_fields'] = ['url'];
```

Comment:
why only url? I see it is the same way in `getUrlWithBasicParameters` but I just do not understand it

Reply:
URL = ID